bindings/rust/src/superchain.rs

```diff
@@ -0,0 +1,300 @@
+    pub chain_id: u64,
+    /// Chain public RPC endpoint
+    pub public_rpc: String,
+    /// Chain sequencer RPC endpoint
+    pub sequencer_rpc: String,
+    /// Chain explorer HTTP endpoint
+    pub explorer: String,
+    /// Level of integration with the superchain.
+    pub superchain_level: SuperchainLevel,
+    /// Time of opt-in to the Superchain.
+    /// If superchain_time is set, hardforks times after superchain_time
+    /// will be inherited from the superchain-wide config.
+    pub superchain_time: Option<u64>,
+    /// Chain-specific batch inbox address
+    pub batch_inbox_addr: Address,
+    /// Chain-specific genesis information
+    pub genesis: ChainGenesis,
+    /// Superchain is a simple string to identify the superchain.
+    /// This is implied by directory structure, and not encoded in the config file itself.
+    #[serde(skip)]
+    pub superchain: String,
+    /// Chain is a simple string to identify the chain, within its superchain context.
+    /// This matches the resource filename, it is not encoded in the config file itself.
+    #[serde(skip)]
+    pub chain: String,
+    #[serde(flatten)]
+    /// Hardfork Configuration. These values may override the superchain-wide defaults.
+    pub hardfork_configuration: HardForkConfiguration,
+    /// Optional Plasma DA feature
+    pub plasma: Option<PlasmaConfig>,
+}
+
+impl ChainConfig {
+    /// Set missing hardfork configurations to the defaults, if the chain has
+    /// a superchain_time set. Defaults are only used if the chain's hardfork
+    /// activated after the superchain_time.
+    pub(crate) fn set_missing_fork_configs(&mut self, defaults: &HardForkConfiguration) {
+        let Some(super_time) = self.superchain_time else {
+            return;
+        };
+        let cfg = &mut self.hardfork_configuration;
+
+        if cfg.canyon_time.is_some_and(|t| t > super_time) {
+            cfg.canyon_time = defaults.canyon_time;
+        }
+        if cfg.delta_time.is_some_and(|t| t > super_time) {
+            cfg.delta_time = defaults.delta_time;
+        }
+        if cfg.ecotone_time.is_some_and(|t| t > super_time) {
+            cfg.ecotone_time = defaults.ecotone_time;
+        }
+        if cfg.fjord_time.is_some_and(|t| t > super_time) {
+            cfg.fjord_time = defaults.fjord_time;
+        }
+    }
+}
+
+/// Block identifier.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct BlockID {
+    /// Block hash
+    pub hash: Hash,
+    /// Block number
+    pub number: u64,
+}
+
+/// Chain genesis information.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ChainGenesis {
+    /// L1 genesis block
+    pub l1: BlockID,
+    /// L2 genesis block
+    pub l2: BlockID,
+    /// Timestamp of the L2 genesis block
+    pub l2_time: u64,
+    /// Extra data for the genesis block
+    pub extra_data: Option<Vec<u8>>,
+    /// Optional System configuration
+    #[serde(flatten)]
+    pub system_config: Option<SystemConfig>,
+}
```

Comment:
Are we able to use alloy's genesis definition here?

https://github.com/alloy-rs/alloy/blob/main/crates/genesis/src/lib.rs#L27

Reply:
I believe that struct holds info about the specific genesis block, whereas this struct is more general and describes an L2 chain genesis. Perhaps the opposite (aka defining ChainGenesis in op-alloy) would make sense?

Reply:
But this type in the superchain-registry looks to match! 
https://github.com/ethereum-optimism/superchain-registry/blob/acf0eef75f26247d7e792ef3f661e54fa9dc73c4/superchain/superchain.go#L700

We can simply re-export it from this module then